### DISCUSSION TITLE
.NET 9 Preview 3

### DISCUSSION THREAD

**Discussion by jamesmontemagno** [dotnet/core#9271] [News]
We are pleased to announce the release of .NET 9 Preview 3. 

Here's a summary of what's new in .NET Libraries, Runtime, & SDK in this preview release:

- [TimeSpan.From overloads](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/libraries.md#adding-timespanfrom-overloads)
- [Added `PersistableAssemblyBuilder` type in System.Reflection.Emit](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/libraries.md#added-persistableassemblybuilder-type-in-systemreflectionemit)
- [Faster exception handling](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/runtime.md#faster-exceptions)
- [Inlining Improvements: Shared Generics with Runtime Lookups](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/runtime.md#inlining-improvements-shared-generics-with-runtime-lookups)
- [Terminal Logger Usability](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/sdk.md#terminal-logger-usability)

For details on these new features and more see the
- [.NET Libraries Release Notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/libraries.md)
- [.NET Runtime Release Notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/runtime.md)
- [.NET SDK Release Notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/sdk.md)

You can find detailed release notes for additional features in .NET 9 Preview 3 below:

* [.NET Data & EF Core](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/efcoreanddata.md)
* [.NET MAUI](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/dotnetmaui.md)
* [ASP.NET Core](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/aspnetcore.md)

You can re-use and re-mix release notes per [license information](https://github.com/dotnet/core/blob/main/release-notes/license-information.md).

## Get started
To get started with .NET 9, [install the .NET 9 SDK](https://dotnet.microsoft.com/download/dotnet/9.0).

If you're on Windows using Visual Studio, we recommend installing the latest [Visual Studio 2022 preview](https://visualstudio.microsoft.com/vs/preview/), or get started with Visual Studio Code and the [C# Dev Kit](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit) extension.

You can find [.NET 9 Preview 3 Installer & Binaries](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview3/9.0.0-preview.3.md) on GitHub.

## Team Announcements & Discussions
Discuss this release with the product teams on GitHub through the announcements for this release. 

* [.NET MAUI](https://github.com/dotnet/maui/discussions/21783) 
* [ASP.NET Core](https://github.com/dotnet/aspnetcore/discussions/55086)
* [Libraries & Runtime](https://github.com/dotnet/runtime/discussions/100928)
* [Containers](https://github.com/dotnet/dotnet-docker/discussions/5351)
* [Source-build](https://github.com/dotnet/source-build/discussions/4320)

## Stay up-to-date with .NET 9
You can stay up-to-date with all the features of .NET 9 with:

* [What's new in .NET 9](https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/overview)
* [What's new in ASP.NET Core](https://learn.microsoft.com/aspnet/core/release-notes/aspnetcore-9.0)
* [What's new in .NET MAUI](https://learn.microsoft.com/dotnet/maui/whats-new/dotnet-9)
* [What's new in EF Core](https://learn.microsoft.com/ef/core/what-is-new/ef-core-9.0/whatsnew)
* [Breaking Changes in .NET 9](https://learn.microsoft.com/dotnet/core/compatibility/9.0)
* [.NET 9 Releases](https://github.com/dotnet/core/blob/main/release-notes/9.0/README.md)

For .NET 9 previews we are utilizing GitHub Discussions to make announcements. For more information [read our discussion](https://github.com/dotnet/core/discussions/9131).

Stay up to date by subscribing to the [RSS feed](https://github.com/dotnet/core/discussions/categories/news.atom) for all release announcements.

You can find a full list of release notes and discussions for all preview releases in the [Release Index](https://github.com/dotnet/core/discussions/9234) thread.

**Comment by masonwheeler** [comment]
> Added `PersistableAssemblyBuilder` type in System.Reflection.Emit

Finally! 🎉  I've been waiting 9 years for this API!  I'll try and get this working, and report back with feedback.

**Reply by RealDoigt** [reply]
At last there's hope for the Boo language yay

**Reply by buyaa-n** [reply]
> I'll try and get this working, and report back with feedback.

Thanks, that would be very helpful. Also, in case you haven't looked into [PDB support proposal](https://github.com/dotnet/runtime/issues/99935) and have feedback please let us know.

**Reply by masonwheeler** [reply]
@buyaa-n Yeah, I've seen that.  That API is definitely going to be part of my tests of this system.

**Comment by SystematicChaos012** [comment]
Will generic UnsafeAccessor be published in Preview 4?

**Reply by jkotas** [reply]
Yes. cc @AaronRobinsonMSFT

**Reply by AaronRobinsonMSFT** [reply]
It is already available in nightly builds. It would be most helpful for people to work with it and let us know about bugs or where the [documentation is insufficient](https://learn.microsoft.com/dotnet/api/system.runtime.compilerservices.unsafeaccessorattribute?view=net-9.0).

**Reply by Jaziel-Ali** [reply]
Uep

**Comment by eder-em** [comment]
@jamesmontemagno please fix this horrible top 1st MOST COMMENTED issue on dotnet MAUI
https://github.com/dotnet/maui/issues/7354

**Comment by InternalHigh** [comment]
Finally faster exceptions. I throw exceptions like there is no tomorrow!

**Comment by alexandrehtrb** [comment]
Please resume posting .NET release news on devblogs :smile:

**Comment by bgrainger** [comment]
Will an api-diff for Preview 3 be published? Preview 2 has one here: https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview2/api-diff/README.md but there's nothing currently available for preview3.

**Reply by martincostello** [reply]
https://github.com/dotnet/core/pull/9273
